kernels/ZendEngine3/main.h

```diff
@@ -285,10 +285,12 @@ int zephir_fetch_parameters(int num_args, int required_args, int optional_args,
 #define ZEPHIR_MM_GET_CONSTANT(return_value, const_name) do { \
 	zval *_constant_ptr = zend_get_constant_str(SL(const_name)); \
 	if (_constant_ptr == NULL) { \
-		ZEPHIR_MM_RESTORE(); \
-		return; \
+	    zval _null; \
+		ZVAL_NULL(&_null); \
+		ZVAL_COPY(return_value, &_null); \
+	} else { \
+	    ZVAL_COPY(return_value, _constant_ptr); \
```

Comment:
Hmm.. What about `ZEPHIR_GET_CONSTANT` then? Also why you removed `ZEPHIR_MM_RESTORE`?

Reply:
Thank you : )

I wanted to do it with minimal change, and now I know that's not enough.

If the constant cannot be detected, it returns directly. So I've now deleted the ZEPHIR_MM_GET_CONSTANT macro.

I want to add this code, but the test case doesn't know how to do it.
```
#define ZEPHIR_GET_CONSTANT(return_value, const_name) do { \
	zval *_constant_ptr = zend_get_constant_str(SL(const_name)); \
	if (_constant_ptr == NULL) { \
	    php_error_docref(NULL, E_WARNING, "Undefined constant %s", const_name); \      <==============
	    zval _null; \
		ZVAL_NULL(&_null); \
		ZVAL_COPY(return_value, &_null); \
	} else { \
	    ZVAL_COPY(return_value, _constant_ptr); \
	} \
} while(0)
```

Test cases can pass.

Is the newly submitted code ok?

Reply:
Because it doesn't return functions directly. ZEPHIR_MM_RESTORE can be executed later. I think so.